### ISSUE TITLE
 Multiple useShortcuts() hooks in the same app?

### ISSUE THREAD

**Issue by chmac** [dagda1/cuttingedge#9]
I tried using `useShortcuts()` in two places, and it looks like only one of them fires. Maybe that's intended.

I guess I'll wire the shortcuts up to redux and only load them in a single place. Would be able to mix local shortcuts on a per route basis with global shortcuts from a wrapping component, but understand this might not be how mousetrap works, etc.

Thanks for this package, made it super easy to add keyboard shorcuts to my app!

**Comment by dagda1** [comment]
@chmac  could you create a codesandbox or supply some code please?  i can fix it then

**Comment by dagda1** [comment]
PRs are welcome also!

**Comment by chmac** [comment]
@dagda1 I had a look at the code, and I think the issue is that in the `useEffect()` cleanup, you call `mousetrap.reset()`. That kills everything listening in mousetrap I think.

In the end, by looking at your code, I realised that using mousetrap directly is pretty easy. I ended up implementing it directly with `useEffect()` like so:

https://github.com/chmac/action.tools/blob/3a6d5f94fc919735a425ae05ff4e8caf4efe7f98/frontend/src/scenes/Bar/Bar.scene.tsx#L31-L38

https://github.com/chmac/action.tools/blob/3a6d5f94fc919735a425ae05ff4e8caf4efe7f98/frontend/src/scenes/Bar/Bar.scene.tsx#L95-L102

By using an ENUM I can also check to make sure I'm handling all the keys.
